internal/netsync/manager.go

```diff
@@ -37,9 +37,31 @@ const (
 	// peer request queue.
 	maxInFlightBlocks = 16
 
-	// maxRejectedTxns is the maximum number of rejected transactions
-	// hashes to store in memory.
-	maxRejectedTxns = 1000
+	// maxRejectedTxns specifies the maximum number of recently rejected
+	// transactions to track.  This is primarily used to avoid wasting a bunch
+	// of bandwidth from requesting transactions that are already known to be
+	// invalid again from multiple peers, however, it also doubles as DoS
+	// protection against malicious peers.
+	//
+	// It is set to target tracking the maximum number of rejected transactions
```

Comment:
Paragraph two is confusing. Suggestion:

```
// Recall that there are 125 connection slots (by default). Assuming the
// default setting of 8 outbound connections (which attackers cannot
// control), that leaves 117 max inbound connections which could potentially be
// malicious. maxRejectedTxns is set to target the maximum number of
// rejected transactions that would result from 120 inbound connections being
// completely malicious. 120 was chosen strictly because it is greater than
// 117, and therefore provides DoS protection.
```

Reply:
The "...and therefore provides DoS protection" line is perhaps overly simplistic...but I wasn't sure if there was a better way to say "120 > 117".

Reply:
Thanks for the suggestion.  I modified it to use this with a few slight modifications as follows:

```Go
	// Recall that there are 125 connection slots by default.  Assuming the
	// default setting of 8 outbound connections, which attackers cannot
	// control, that leaves 117 max inbound connections which could potentially
	// be malicious.  maxRejectedTxns is set to target tracking the maximum
	// number of rejected transactions that would result from 120 connections
	// with malicious peers.  120 is used since it is strictly greater than the
	// aforementioned 117 max inbound connections while still providing for the
	// possibility of a few happenstance malicious outbound connections as well.
```